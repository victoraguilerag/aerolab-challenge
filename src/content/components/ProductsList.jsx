import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  updateSelectedItem,
  redeemProduct
} from '../../actions/index.js'
import CSSTransitionGroup from 'react-addons-css-transition-group'
import '../../animations/animations.css'
import { connect } from 'react-redux'
import Product from './Product'

export class ProductsList extends Component {
  handleProductClick = (id) => {
    if (this.props.selectedItem === id) {
      return this.props.updateSelectedItem('')
    }
    this.props.updateSelectedItem(id)
  }
  handleRedeemProduct = id => {
    this.props.redeemProduct(id)
  }
  render () {
    const {
      products,
      coins,
      handleModal,
      selectedItem
    } = this.props
    const productsComponents = products.map((product, index) => {
      const { _id: id } = product
      const selected = (selectedItem === id)

      return (
        <CSSTransitionGroup
          transitionName="product"
          transitionEnterTimeout={1500}
          transitionLeaveTimeout={2}
          style={styles.productList}
          key={id}
        >
          <Product
            info={product}
            selected={selected}
            coins={coins}
            handleProductClick={this.handleProductClick}
            handleRedeemProduct={this.handleRedeemProduct}
            handleModal={handleModal}
          />
        </CSSTransitionGroup>
      )
    })
    return (
      <div className="product-list" style={styles.productList}>
        {
          productsComponents
        }
      </div>
    )
  }
}

const styles = {
  'productList': {
    'display': 'flex',
    'flexFlow': 'row wrap',
    'justifyContent': 'space-around',
    'maxWidth': '1176px',
    'margin': '0 auto'
  }
}

ProductsList.propTypes = {
  selectedItem: PropTypes.string.isRequired,
  updateSelectedItem: PropTypes.func.isRequired,
  redeemProduct: PropTypes.func.isRequired,
  products: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  coins: PropTypes.number.isRequired,
  handleModal: PropTypes.func.isRequired
}

const mapStateToProps = (state) => {
  return {
    selectedItem: state.selectedItem
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    updateSelectedItem: id => dispatch(updateSelectedItem(id)),
    redeemProduct: id => dispatch(redeemProduct(id))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ProductsList)
