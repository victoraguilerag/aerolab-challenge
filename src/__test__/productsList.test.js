import React from 'react'
import { ProductsList } from '../content/components/ProductsList.jsx'
import productsStub from '../stub/products.js'
import renderer from 'react-test-renderer'

const selectedItem = '123123'
const selected = true
const products = []
const coins = 123
const info = productsStub[0]
const functionMock = () => console.log('mock')

test('Product List renders', () => {
  const component = renderer.create(
    <ProductsList
      selectedItem={selectedItem}
      updateSelectedItem={functionMock}
      redeemProduct={functionMock}
      products={products}
      coins={coins}
      handleModal={functionMock}
    />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
