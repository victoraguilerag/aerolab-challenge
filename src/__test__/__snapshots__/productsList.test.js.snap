// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product List renders 1`] = `
<div
  className="product-list"
  style={
    Object {
      "display": "flex",
      "flexFlow": "row wrap",
      "justifyContent": "space-around",
      "margin": "0 auto",
      "maxWidth": "1176px",
    }
  }
/>
`;
